apps/payment/signals.py

```diff
@@ -0,0 +1,34 @@
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+
+from . import status
+from .models import PaymentRelation, PaymentTransaction
+
+
+@receiver(signal=pre_save, sender=PaymentRelation)
+def handle_payment_relation_status_change(sender, instance: PaymentRelation, **kwargs):
+    if instance.status == status.SUCCEEDED:
+        # Handle the completed payment. Remove delays, suspensions and marks
+        instance.payment.handle_payment(instance.user)
+        instance.status = status.DONE
+    elif instance.status == status.REFUNDED:
+        instance.refunded = True
+        instance.payment.handle_refund(instance)
+        instance.status = status.REMOVED
+
+
+@receiver(signal=pre_save, sender=PaymentTransaction)
+def handle_payment_transaction_status_change(sender, instance: PaymentTransaction, **kwargs):
+    # When a payment succeeds, ot should be stored to the DB """
```

Comment:
You missed the closing quotations :stuck_out_tongue:

Reply:
:upside_down_face: